docs/array-utils.md

```diff
@@ -33,4 +33,13 @@ Return a copy of the given array without falsy values (eg: false, 0, '', null, u
 ```typescript
 const array = ['', false, null, 'valid', 1, undefined, 0]
 console.log(removeFalsy(array)) // result: ['valid', 1]
-```
\ No newline at end of file
+```
+
+
+
+`keyArrayBy<T extends { [K in keyof T]: RecordKeyType }, K extends keyof T>(array: T[], selector: K, arrayValue: boolean = true): Record<T[K], T | T[]>`
+
+Creates a Record composed of keys from `selector`. 
+The corresponding value of each key is:
+- If `arrayValue` is `true` an array of all elements with the same `selector`.
```

Comment:
we have groupBy for storing array values, doesn't it do the same thing? not sure if we need to support here as well

Reply:
Mmm good catch! I don’t how I missed it, I think we need both options but instead of having a new method I will try to extend groupBy to support both, if that makes sense

Reply:
But in that case maybe groupBy is no a good name 😅, I am thinking

Reply:
yeah, I suggest having two separate methods with clear purpose

Reply:
As discussed on slack, for visibility, groupBy doesn't offer some type of features really useful so we decided to modify it to offer them